### PR TITLE
fix(init-templates): numerous fixes and additions to the sample-app init templates

### LINKED ISSUE #4418: Review & Harmonize CDK init templates
We should make sure all the init templates we have across the languages are consistent (same set of templates across all the languages, with similar/comparable content).

### LINKED ISSUE #5130: csharp init template: Possible unassigned object created by 'new'
When creating a new csharp project, it comes with a code analysis issue "RECS0026" of "Possible unassigned object created by 'new'". This is because constructs are often not assigned to values.

### Reproduction Steps

```
cdk init -l csharp
```

Open in Visual Studio.

### Solution

We should include a `GlobalSuppressions.cs` file with:

```csharp
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Potential Code Quality Issues", "RECS0026:Possible unassigned object created by 'new'", Justification = "Constructs add themselves to the scope in which they are created")]
```

### Environment

  - **CLI Version      :** 1.17.1
  - **Framework Version:** 1.17.1
  - **OS               :** Mac
  - **Language         :** C#

--- 

This is :bug: Bug Report

### LINKED ISSUE #5131: csharp init template: stack props should be optional
The current init template includes a generated stack class where the stack props are required and its definition passes an empty object. That's bad form.

Instead of:

```csharp
public MyTestStack(Construct scope, string id, IStackProps props) : base(scope, id, props)
```

And:

```csharp
new MyTestStack(app, "MyTestStack", new StackProps());
```

We want:

```csharp
public MyTestStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
```

and:

```csharp
new MyTestStack(app, "MyTestStack");
```

### Reproduction Steps

```
cdk init -l csharp
```

### Environment

  - **CLI Version      :** 1.17.1
  - **Framework Version:** 1.17.1
  - **OS               :** Mac
  - **Language         :** C#

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by NGL321**
Fixes #5131 
Fixes #4418 
Fixes #5130

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/init-templates/app/csharp/src/%name.PascalCased%/%name.PascalCased%.template.csproj [hunk 1]
```diff
@@ -8,10 +8,9 @@
   <ItemGroup>
     <!-- CDK Construct Library dependencies -->
     <PackageReference Include="Amazon.CDK" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.IAM" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.SNS" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.SNS.Subscriptions" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.SQS" Version="%cdk-version%-devpreview" />
+
+    <!-- jsii Roslyn analyzers -->
+    <PackageReference Include="Amazon.Jsii.Analyzers" Version="*" PrivateAssets="all" />
```
Comment (eladb) [inline packages/aws-cdk/lib/init-templates/app/csharp/src/%name.PascalCased%/%name.PascalCased%.template.csproj]: Are we again trying this out? I would vote against introducing this now.

Comment (RomainMuller) [reply packages/aws-cdk/lib/init-templates/app/csharp/src/%name.PascalCased%/%name.PascalCased%.template.csproj]: Making them into a comment instead.

**Comment by RomainMuller** [comment]
Confirming successful local execution of the `test/integ/init/test-all.sh` suite (`C#`, `F#`, `Python`, `JS`, `TS`).

File: packages/aws-cdk/lib/init-templates/app/csharp/src/%name.PascalCased%/%name.PascalCased%.template.csproj [hunk 2]
```diff
@@ -8,10 +8,10 @@
...
     <!-- CDK Construct Library dependencies -->
     <PackageReference Include="Amazon.CDK" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.IAM" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.SNS" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.SNS.Subscriptions" Version="%cdk-version%-devpreview" />
-    <PackageReference Include="Amazon.CDK.AWS.SQS" Version="%cdk-version%-devpreview" />
+
+    <!-- jsii Roslyn analyzers (un-comment to obtain compile-time checks for missing required props
+    <PackageReference Include="Amazon.Jsii.Analyzers" Version="*" PrivateAssets="all" />
+    -->
```
Comment (eladb) [inline packages/aws-cdk/lib/init-templates/app/csharp/src/%name.PascalCased%/%name.PascalCased%.template.csproj]: nice touch

File: packages/aws-cdk/lib/init-templates/sample-app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.myorg;
+
+import software.amazon.awscdk.core.App;
+
+public final class %name.PascalCased%App {
+    public static void main(final String argv[]) {
```
Comment (shivlaks) [inline packages/aws-cdk/lib/init-templates/sample-app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java]: args?

File: packages/aws-cdk/lib/init-templates/sample-app/fsharp/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+
+# CDK HelloWorld
+
+Welcome to your CDK .NET project!
```
Comment (shivlaks) [inline packages/aws-cdk/lib/init-templates/sample-app/fsharp/README.md]: do we want to call out F# at any point in the README?

File: packages/aws-cdk/lib/init-templates/sample-app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java (hunk 3 above)
Comment (skinny85) [reply packages/aws-cdk/lib/init-templates/sample-app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java]: Actually, this brackets style (`String argv[]`) causes a warning in Intellij: "C-style array declaration". The more idiomatic Java way is to say `String[] args`.

Comment (shivlaks) [reply packages/aws-cdk/lib/init-templates/sample-app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java]: @skinny85 yes that's what I meant by that comment. might just pull this down and make that change myself

File: packages/aws-cdk/lib/init-templates/app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java [hunk 5]
```diff
@@ -5,7 +5,7 @@
 import java.util.Arrays;
 
 public class %name.PascalCased%App {
-    public static void main(String[] args) {
+    public static void main(final String[] args) {
```
Comment (RomainMuller) [inline packages/aws-cdk/lib/init-templates/app/java/src/main/java/com/myorg/%name.PascalCased%App.template.java]: `static`s in Java cannot je overloaded so one can consider than `static` implies `final`. From this perspective, I would not explicitly mark it (I have known linters that flag this as a problem)

**Pull request merged**